### PR TITLE
docs: updated CONTRIBUTING.md

### LINKED ISSUE #711: docs: update `CONTRIBUTING.md`
**Description**

As a developer interested in contributing I would like to have a `CONTRIBUTING.md` file that outlines:
1. How to run the tool in my local environment
2. How the overall structure of the tool works (explained with visualisations)
3. Know on what features the team would like external contributors to work and how to create a correct pull request.

**Requirements**

- [ ] Explain how the tool works internally (with visualisations)
- [ ] Include instructions on how to run the tool in my local environment (for MacOs, Windows and Linux)
- [ ] Include instructions on how to find features available for contribution or suggest features
- [ ] Include instructions on how to open a good pull request or issue

### PR DESCRIPTION
**Pull request by alestiago**
## Description

Related to #711 (partially resolves).

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [X] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by scarletteliza** [comment]
made some changes!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,31 +1,129 @@
...
+
+When reporting a bug, please use the built-in [Bug Report][bug_report_link] template and provide as much information as possible including detailed reproduction steps. Once one of the package maintainers has reviewed the issue and we reach an agreement on the fix, open a pull request.
+
+## Developing for Very Good CLI
+
+To develop for Very Good CLI you will need to become familiar with Very Good Ventures processes and conventions:
+
+### Setting up your local development environment
+
+1. Install a valid [Dart SDK](https://dart.dev/get-dart) in your local environment. Compatible Dart SDK versions with test optimizer can be found [here](https://github.com/VeryGoodOpenSource/very_good_cli/blob/66c2a5e847ba4db6a9cb212e2b08e2af9f9e70d2/pubspec.yaml#L7). If you have Flutter installed, you likely have a valid Dart SDK version already installed.
```
Comment (wolfenrain) [inline CONTRIBUTING.md:36]: 👀  should this not refer to main?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,31 +1,129 @@
...
+```sh
+# Automatically fix analysis issues that have associated automated fixes
+dart fix --apply
+```
+
+> **Note**: Our repositories use [Very Good Analysis](https://github.com/VeryGoodOpenSource/very_good_analysis). 
+
+8. Create the Pull Request with a meaningful description, linking to the original issue where possible.
+
+9. Verify that all [status checks](https://github.com/VeryGoodOpenSource/very_good_cli/actions/) are passing.
```
Comment (wolfenrain) [inline CONTRIBUTING.md:124]: If it is a fork, the actions wont live on our repo right? Also the actions will only run if we approve them to run for first-time contributors

Comment (alestiago) [reply CONTRIBUTING.md]: I've changed this, let me know if you're happy with the new change by resolving this comment.

File: CONTRIBUTING.md (hunk 1 above)
Comment (alestiago) [reply CONTRIBUTING.md]: I've changed this, let me know if you're happy with the new change by resolving this comment.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,31 +1,129 @@
...
-1. Fork the repository and create your branch from `main`.
-1. Install all dependencies (`flutter packages get` or `pub get`).
-1. Squash your commits and ensure you have a meaningful, [semantic][conventional_commits_link] commit message.
-1. Add tests! Pull Requests without 100% test coverage will not be approved.
-1. Ensure the existing test suite passes locally.
-1. Format your code (`dart format .`).
-1. Analyze your code (`dart analyze --fatal-infos --fatal-warnings .`).
-1. Create the Pull Request.
-1. Verify that all status checks are passing.
+💡 Note: Very Good CLI's completion functionality is powered by [CLI Completion](https://github.com/VeryGoodOpenSource/cli_completion) and its test command optimization is powered by [test optimizer](bricks/test_optimizer/README.md). If you want to contribute to either of those, please refer to their respective repositories.
```
[the remark this answers is not in the document]
Comment (alestiago) [reply CONTRIBUTING.md]: I've changed this, let me know if you're happy with the new change by resolving this comment.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,31 +1,137 @@
...
+
+# Or get project dependencies manually
+dart pub get && cd bricks/test_optimizer && dart pub get && cd ../../
+```
+
+3. Run all Very Good CLI tests:
+
+```sh
+# 💻 Run Very Good CLI's unit tests
+dart test
```
[the remark this answers is not in the document]
Comment (alestiago) [reply CONTRIBUTING.md:52]: Does running `dart test` succeed in your environment out of the box?

[the remark this answers is not in the document]
Comment (wolfenrain) [reply CONTRIBUTING.md:52]: No idea, if it doesnt for you, you should open an issue 😉

**Pull request merged**